src/GT.Core/Services/Impl/GTCompanyService.cs

```diff
@@ -64,26 +65,87 @@ public async Task<CompanyDTO> AddAsync(CompanyDTO dto)
 			}
 		}
 
-		public string AddCompanyLogo(byte[] file)
+		public async Task AddCompanyLogo(CompanyLogoDTO companyLogoDTO)
 		{
-			throw new NotImplementedException();
+			//create folder if not exist
+			string path = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot/img/CompanyLogos");
+			if (!Directory.Exists(path))
+				Directory.CreateDirectory(path);
+
+			if (companyLogoDTO.File == null || companyLogoDTO.CompanyId == null)
+			{
+				_logger.LogWarning($"Can not use null arguments in method: {nameof(AddCompanyLogo)}");
+				return;
+			}
+
+			var company = await GetByIdAsync(companyLogoDTO.CompanyId);
+			if (company == null)
+			{
+				_logger.LogWarning($"Could not find a company in method: {nameof(AddCompanyLogo)}");
+				return;
+			}
+
+			// Check if company already have a stored logo
```

Comment:
Suggested change:
// Check if company already has a stored logo

Reply:
Better, I've changed to the suggested comment